Converted the table to react

kharann opened the pull request:
React table allows easier column filtering/sorting by pressing the table headers. This works i believe. 

I dunno how to fix the linting issues which is left. Some kind of props validation error. Posted pull request for some help/guidance forward on how to handle a few issues/get some feedback on what to improve. Just need som advice :)
EDIT: Small issue with z-indexing the end genfors prompt since react doesnt support z-index i guess

sklirg commented:
I'm not familiar with react tables, so tbh I'm not suited to review this code.

React does noting with z indexing, z index is a styling thing and not react. So that sounds like CSS to me.

kharann commented:
> I'm not familiar with react tables, so tbh I'm not suited to review this code.
> 
> React does noting with z indexing, z index is a styling thing and not react. So that sounds like CSS to me.

Seems like i was just dumb and forgot to stop and restart yarn frontend since it have to reload the css modules. Works now hehe. Was super confused since the site just kept ignoring new properties i added

kharann commented:
Some screenshots too:

Normal view of a person who cannot vote:
![screenshot_2019-03-07_14-29-12](https://user-images.githubusercontent.com/41551253/53960058-87b53900-40e5-11e9-903d-7a7ef84fa729.png)

Warning popup still works if you're about to allow a user to vote which doesnt have the permissions from the start
![screenshot_2019-03-07_14-29-29](https://user-images.githubusercontent.com/41551253/53960060-87b53900-40e5-11e9-88bc-73947f8b4ea4.png)

Color changes as normal are allowed to vote
![screenshot_2019-03-07_14-29-46](https://user-images.githubusercontent.com/41551253/53960061-87b53900-40e5-11e9-8150-8fb26313f620.png)

The sorting mechanisms on the header works as follows:

Name: sorts alphabetically with focus on first name. For example Andre Bob would come before Bob Andre. Either descending or ascending ordier

Registrert: Sorts users who are registered first or not registered first depending on ascending or descending

Rettigheter: Sorts users depending on their permission level where Tellekorps is (maybe) the highest level while Ingen is the lowest. For example Tellekorps, Stemmerett ..., ... , Ingen,.

Stemmeberttiggelser: Sorts users with green toggle/red toggle/the lock toggle

duvholt commented on client/src/components/Admin/Users/List/index.js:
```diff
@@ -19,42 +23,73 @@ const UserList = ({ users, toggleCanVote }) => {
+    Header: `Registrert (${usersRegistered})`,
+    accessor: 'registered',
+    className: css.center,
+    Cell: props =>
```
To fix the props linting I would extract this inline component. 
IMO it's fine to just add another component to this file just as long as you don't put it inside the render function.

duvholt commented on client/src/components/Admin/Users/List/index.js:
```diff
@@ -19,42 +23,73 @@ const UserList = ({ users, toggleCanVote }) => {
+    accessor: 'canVote',
+    className: css.center,
+    Cell: props =>
+      <ToggleCanVoteContainer
```
Same as the other prop types comment.

duvholt commented on client/src/components/Admin/Users/List/index.js:
```diff
@@ -19,42 +23,73 @@ const UserList = ({ users, toggleCanVote }) => {
+      data={data}
+      columns={columns}
+      className={css.table}
+      getTrProps={(state, rowInfo, column, instance) => ({
```
`instance` is unused here so you can just remove it.

kharann replied:
Seems like i could remove instance and column. Reason was i couldnt remove state else it wouldnt render even if its unused due to the syntax for getTrTrops

kharann commented:
Solved all the issues! I cleaned up the List/index file a little so it looks a lot cleaner. Also moved each of the cell components to another Table.js and defined the propTypes to remove all the linting issues

torjusti commented on client/src/components/Admin/Users/List/List.css:
```diff
@@ -1,32 +1,57 @@
-  margin-top: 1rem;
-  border-collapse: collapse;
+.table {
+  margin-top: 25px;
```
I think it would be best if we stuck to REM everywhere

torjusti commented on client/src/components/Admin/Users/List/Table.js:
```diff
@@ -0,0 +1,51 @@
+        [css.close]: !props.value.isRegistered,
+        [css.toggle]: props.value.isRegistered,
+      },
+        )}
```
This indenting feels a bit off

torjusti commented on client/src/components/Admin/Users/List/index.js:
```diff
@@ -19,42 +22,63 @@ const UserList = ({ users, toggleCanVote }) => {
     .length;
+
+  const data = [...Object.keys(users)
+    .sort((a, b) => users[a].name.localeCompare(users[b].name))
```
I don't think this is necessary anymore now that we're using react table?
